gensim/test/test_matutils.py

```diff
@@ -0,0 +1,187 @@
+        rs = self.random_state
+
+        for dtype in [np.float16, np.float32, np.float64]:
+            for i in range(self.num_runs):
+                input1 = rs.uniform(-10000, 10000, size=(self.num_topics,))
+                input2 = rs.uniform(-10000, 10000, size=(self.num_topics,))
+
+                known_good = mean_absolute_difference(input1, input2)
+                test_values = matutils.mean_absolute_difference(input1, input2)
+
+                msg = "mean_absolute_difference failed for dtype={}".format(dtype)
+                self.assertTrue(np.allclose(known_good, test_values), msg)
+
+    def testDirichletExpectation(self):
+        # test dirichlet_expectation
+        rs = self.random_state
+
+        for dtype in [np.float16, np.float32, np.float64]:
+            for i in range(self.num_runs):
+                # 1 dimensional case
+                input_1d = rs.uniform(.01, 10000, size=(self.num_topics,))
+                known_good = dirichlet_expectation(input_1d)
+                test_values = matutils.dirichlet_expectation(input_1d)
+
+                msg = "dirichlet_expectation_1d failed for dtype={}".format(dtype)
+                self.assertTrue(np.allclose(known_good, test_values), msg)
+
+                # 2 dimensional case
+                input_2d = rs.uniform(.01, 10000, size=(1, self.num_topics,))
+                known_good = dirichlet_expectation(input_2d)
+                test_values = matutils.dirichlet_expectation(input_2d)
+
+                msg = "dirichlet_expectation_2d failed for dtype={}".format(dtype)
+                self.assertTrue(np.allclose(known_good, test_values), msg)
+		
+class UnitvecTestCase(unittest.TestCase):
+
+    def manual_unitvec(self, vec):
+	self.vec = vec
+	self.vec = self.vec.astype(np.float)
+	if sparse.issparse(self.vec):
+	    vec_sum_of_squares = self.vec.multiply(self.vec)
+            unit = 1. / np.sqrt(vec_sum_of_squares.sum())
+	    return self.vec.multiply(unit)
+        elif not sparse.issparse(self.vec):
+	    sum_vec_squared = np.sum(self.vec ** 2)
+            self.vec /= np.sqrt(sum_vec_squared)
+	    return self.vec
+
+    def test_inputs(self):
+	input_dtypes = [np.float32, np.float64, np.int32, np.int64]
+	input_arrtypes = ['sparse', 'normal']
+	for dtype_ in input_dtypes:
+	    for arrtype in input_arrtypes:
+		if arrtype == 'normal':
+    		    if dtype_ == np.float32 or dtype_ == np.float64:
+			input_vector = np.random.uniform(size=(5,)).astype(dtype_)
+			unit_vector = unitvec_with_bug.unitvec(input_vector)
+			man_unit_vector = self.manual_unitvec(input_vector)
+			self.assertEqual(input_vector.dtype, unit_vector.dtype)
+			self.assertTrue(np.allclose(unit_vector, man_unit_vector))
+    		    else:
+		        input_vector = np.random.randint(10, size=5).astype(dtype_)
+			unit_vector = unitvec_with_bug.unitvec(input_vector)
+			man_unit_vector = self.manual_unitvec(input_vector)
+			self.assertTrue(np.allclose(unit_vector, man_unit_vector))
```

Comment:
you should check the dtype for this case too (you know that this will be default)